server/api/api.go

```diff
@@ -114,12 +118,85 @@ func (h *Handler) setLink(w http.ResponseWriter, r *http.Request) {
 	status := http.StatusNotModified
 	if changed {
 		if err := h.store.SaveLinks(links); err != nil {
-			h.handleError(w, errors.Wrap(err, "unable to save link"))
+			h.handleError(w, errors.Wrap(err, "unable to save the link"))
 			return
 		}
 		status = http.StatusOK
 	}
 
+	ReturnStatusOK(status, w)
+}
+
+func (h *Handler) deleteLink(w http.ResponseWriter, r *http.Request) {
+	autolinkName := r.URL.Query().Get(autolinkclient.AutolinkNameQueryParam)
+	if autolinkName == "" {
+		h.handleError(w, errors.New("autolink name or pattern should not be empty"))
+		return
+	}
+
+	links := h.store.GetLinks()
+	found := false
+	for i := 0; i < len(links); i++ {
+		if links[i].Name == autolinkName {
+			links = append(links[:i], links[i+1:]...)
+			found = true
+		}
+	}
+
+	status := http.StatusNotModified
+	if found {
+		if err := h.store.SaveLinks(links); err != nil {
+			h.handleError(w, errors.Wrap(err, "unable to save the link"))
+			return
+		}
+		status = http.StatusOK
+	}
+
+	ReturnStatusOK(status, w)
+}
+
+func (h *Handler) getLinks(w http.ResponseWriter, r *http.Request) {
+	links := h.store.GetLinks()
+
+	autolinkName := r.URL.Query().Get(autolinkclient.AutolinkNameQueryParam)
+	if autolinkName == "" {
+		h.handleSendingJSONContent(w, links)
+		return
+	}
+
+	found := false
+	var autolinks []autolink.Autolink
+	for _, link := range links {
+		if link.Name == autolinkName {
+			autolinks = append(autolinks, link)
+			found = true
+		}
+	}
+
+	if !found {
+		h.handleError(w, errors.Errorf("no autolink found with name %s", autolinkName))
+		return
+	}
```

Comment:
If we're only matching by name and the name is unique, we're only going to find one link per query, right? In that case, this could be something like:

```suggestion
	var autolink *autolink.Autolink = nil
	for _, link := range links {
		if link.Name == autolinkName {
			autolink = link
			break
		}
	}

	if autolink == nil {
		h.handleError(w, errors.Errorf("no autolink found with name %s", autolinkName))
		return
	}
```

And then we should change the client to match the single response, see comments there.